### ISSUE TITLE
Exec: RootDirectory in combination with DynamicUser and StateDirectory

### ISSUE THREAD

**Issue by aw32** [systemd/systemd#8965]
**systemd version the issue has been seen with**
> systemd 238.76-1  (f398c546c6fc43121131f41acec56b5a851bd35e)

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > The symlink /var/lib/service -> /var/lib/private/service should have been set.
 > Resolving of user name/id should work.

**Unexpected behaviour you saw**
 > The symlink is created in the host filesystem, but not in the filesystem of the service (as specified by RootDirectory). However the /var/lib/private/service directory is present.
 > -> This can be fixed manually by adding the expected symlink in the filesystem specified in RootDirectory.
 > Unexpectedly the user name/id resolving does not work. The reason is the missing dbus socket and/or missing entries in the user configuration.
 > -> This can be fixed manually by adding BindPaths=/run/dbus:/run/dbus: to make dbus available to the service.
 > -> Entries in the user configuration can't be added manually, because of the dynamic nature of the dynamic user id.
 >
 > Without RootDirectory this works, because the symlink and the dbus socket are present in the host filesystem.
 > I don't know if this should be fixed, but a hint concerning this combination in the man page might be helpful. (In case there is already a hint that I overlooked: nevermind.)
  

**Steps to reproduce the problem**
Example unit with RootDirectory, DynamicUser and StateDirectory entries.
```
[Unit]
Description=Test dir service

[Service]
User=testdir
Group=testdir
ExecStart=/usr/bin/ls -la /var/lib/                                                                                                             
RootDirectory=/path/to/root/
DynamicUser=true
StateDirectory=testdir

[Install]
WantedBy=multi-user.target
```
/path/to/root/ should contain expected content for a root filesystem.
In my case created using the following command as described in the [Arch Wiki](https://wiki.archlinux.org/index.php/Systemd-nspawn#Create_and_boot_a_minimal_Arch_Linux_distribution_in_a_container) to create a minimal Arch system:
```
pacstrap -i -c /path/to/root/ base --ignore linux 
```

**Comment by poettering** [comment]
This should definitely work. if it doesn't it's a bug.

**Comment by yuwata** [comment]
Fix is waiting in #9084.
